### ISSUE TITLE
Command error Panic: AzcopyAppPathFolder should be initialized by root

### ISSUE THREAD

**Issue by namanmeena** [Azure/azure-storage-azcopy#1334]
##### Note: The version is visible when running AzCopy without any argument
Version: 10.8.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Platform: Linux

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.
`azcopy login --service-principal --application-id {some_app_id} --tenant-id {some_tenant_id}`

### What problem was encountered?
The command was ran through spark-submit in a python file using `subprocess` package. It is running fine in client mode and without spark-submit but while running it in spark's cluster mode, I'm getting the following error message -
```
panic: invalid state, azcopyAppPathFolder should be initialized by root

```

### How can we reproduce the problem in the simplest way?
Run the azcopy login command in a python file using subprocess with spark-submit in cluster mode.
You can use the command below -
`azcopy login --service-principal --application-id {some_app_id} --tenant-id {some_tenant_id}`
or
`azcopy login`

### Have you found a mitigation/solution?
No.

**Comment by namanmeena** [comment]
Resolved it. 

It requires the `.azcopy` directory in the user home directory (`/home/username` in linux). If not present, it creates that directory, and upon failure to do so will throw the above error.

It was trying to create directory `/home/naman/.azcopy` and failing since it was running in yarn container and didn't had admin privileges.

**Comment by kevsor2** [comment]
I'm trying to use AzCopy in an Azure PowerShell Function and hitting this error.  The Azure functions don't have AzCopy natively installed, so I've already had to get it loaded and unzipped, and then set the log and plan paths based on these entries:
https://gist.github.com/andrew-kelleher/031ff5daa8849a9a9c335271f0b3bc08
https://github.com/Azure/azure-storage-azcopy/issues/1166

But I've tried several locations for the executable and log/plan paths, and am still getting the "panic: invalid state, azcopyAppPathFolder should be initialized by root" error when I try to login in AzCopy.

Any ideas?
